File: java/src/main/java/it/giannibombelli/wsc2026/giftcard/domain/events/GiftCardRefunded.java
package it.giannibombelli.wsc2026.giftcard.domain.events;

import it.giannibombelli.wsc2026.common.utils.Require;

import it.giannibombelli.wsc2026.common.domain.primitive.Money;
import it.giannibombelli.wsc2026.giftcard.domain.giftcard.GiftCardId;

public record GiftCardRefunded(
    GiftCardId aggregateId,
    Money refundedAmount
) implements GiftCardEvent {
    public GiftCardRefunded {
        Require.requireArgument(aggregateId, "giftCardId");
        Require.requireArgument(refundedAmount, "refundedAmount");
    }
}

File: typescript/src/giftcard/domain/events/giftCardRefunded.ts
// Evento emesso quando una gift card viene rimborsata.

import type { Event } from '@/common/domain/model/event.js';
import type { Money } from '@/common/domain/primitive/money.js';
import type { GiftCardId } from '../giftcard/giftCardId.js';
import { requireArgument } from '@/common/utils/requireArgument.js';

export type GiftCardRefunded = Event<'GiftCardRefunded', GiftCardId> & {
  readonly refundedAmount: Money;
};

export function giftCardRefunded(aggregateId: GiftCardId, refundedAmount: Money): GiftCardRefunded {
  requireArgument(aggregateId, 'giftCardId');
  requireArgument(refundedAmount, 'refundedAmount');
  return { kind: 'GiftCardRefunded', aggregateId, refundedAmount };
}
